atomic-factory/src/test/java/org/infinispan/atomic/sharded/collections/ShardedTreeMapTest.java

```diff
@@ -0,0 +1,90 @@
+package org.infinispan.atomic.sharded.collections;
+
+import org.infinispan.Cache;
+import org.infinispan.atomic.AtomicObjectFactory;
+import org.infinispan.configuration.cache.CacheMode;
+import org.infinispan.configuration.cache.ConfigurationBuilder;
+import org.infinispan.manager.EmbeddedCacheManager;
+import org.infinispan.test.AbstractCacheTest;
+import org.infinispan.test.MultipleCacheManagersTest;
+import org.infinispan.test.TestingUtil;
+import org.infinispan.test.fwk.TransportFlags;
+import org.testng.annotations.Test;
+
+import java.util.*;
+
+
+/**
+ *
+ * @author otrack
+ * @since 7.0
+ */
+@Test(groups = "functional", testName = "ShardedTreeMapTest")
+public class ShardedTreeMapTest extends MultipleCacheManagersTest {
+
+    private static int NCALLS= 100;
+    private static int NCACHES = 3;
+    private static List<Cache> caches = new ArrayList<Cache>();
+
+    @Test(enabled = true)
+    public void basicUsageTest() throws  Exception{
+        EmbeddedCacheManager cacheManager = cacheManagers.iterator().next();
+        Cache cache = cacheManager.getCache();
+        AtomicObjectFactory factory = new AtomicObjectFactory(cache);
+
+        SortedMap<Integer,Integer> map = factory.getInstanceOf(ShardedTreeMap.class,"test",false,null,false,3);
+        for(int i=0; i<NCALLS; i++) {
+            map.put(i,i);
+            map.get(i);
+        }
+        factory.disposeInstanceOf(ShardedTreeMap.class,"test",true);
+        map = factory.getInstanceOf(ShardedTreeMap.class,"test",false,null,false,3);
+        int a = map.size();
+        log.debug(a);
+        assert map.size() == NCALLS;
```

Comment:
We prefer the use of the AssertJUnit static assert methods, since they provide better messages